### DISCUSSION TITLE
How to correctly transmit error details to client with gRPC-Json transcoding?

### DISCUSSION THREAD

**Discussion by Sayan751** [dotnet/aspnetcore#49179] [Q&A] [answered]
I have a gRPC service with JSON transcoding enabled. An endpoint of the service throws `RpcException` with same status code but different metadata. Examples:

```csharp
throw new RpcException(new Status(StatusCode.InvalidArgument, "Whatever1"), new Metadata
{
    { "Code", Errors.Error1.ToString() }
});

throw new RpcException(new Status(StatusCode.InvalidArgument, "Whatever2"), new Metadata
{
    { "Code", Errors.Error2.ToString() }
});
```

Whoever, when I am making an HTTP call to the service, then I am getting the following:

```json
{
  "code": 3,
  "message": "Whatever1",
  "details": []
}
```

I expected the trailers to appear in the `details` array. Am I missing something here? At the end, I need a way to differentiate between the two errors with the same status code. Is that possible?

**Comment by Xyncgas** [comment]
that's weird, different error message should be sent to you already, in the header

**Comment by gfoidl** [comment, marked as answer]
Let's continue in https://github.com/dotnet/aspnetcore/issues/49196
